subxt/src/storage/storage_key.rs

```diff
@@ -0,0 +1,232 @@
+    }
+    let start_idx = address_bytes.len() - cursor.len();
+    if let Err(err) = scale_decode::visitor::decode_with_visitor(
+        cursor,
+        ty_id,
+        metadata.types(),
+        scale_decode::visitor::IgnoreVisitor,
+    ) {
+        return Some(Err(scale_decode::Error::from(err).into()));
+    }
+    let end_idx = address_bytes.len() - cursor.len();
+    let key_bytes = address_bytes[start_idx..end_idx].to_vec();
+    let key = StaticStorageKey {
+        bytes: Static(Encoded(key_bytes)),
+        _marker: std::marker::PhantomData::<K>,
+    };
+    Some(Ok(key))
+}
+
+impl StorageKey for Vec<scale_value::Value> {
+    fn keys_iter(&self) -> impl ExactSizeIterator<Item = &dyn EncodeAsType> {
+        // Note: this returns the storage root address of the storage entry.
+        // It gives the same result as if you were to use `vec![]` as a `StorageKey`.
+        self.iter().map(|e| e as &dyn EncodeAsType)
+    }
+
+    fn decode_from_address_bytes(
+        address_bytes: &[u8],
+        hashers_and_ty_ids: &[(StorageHasher, u32)],
+        metadata: &Metadata,
+    ) -> Option<Result<Self, Error>>
+    where
+        Self: Sized + 'static,
+    {
+        let cursor = &mut &*address_bytes;
+        if let Err(err) = strip_storage_addess_root_bytes(cursor) {
+            return Some(Err(err.into()));
+        }
+        let mut hashers_and_ty_ids_iter = hashers_and_ty_ids.iter();
+
+        let mut result: Vec<scale_value::Value> = vec![];
+        while !cursor.is_empty() {
+            let Some((hasher, ty_id)) = hashers_and_ty_ids_iter.next() else {
+                // Still bytes left, but no hashers and type ids anymore to pull from: this is an unexpected error.
+                return Some(Err(StorageAddressError::UnexpectedAddressBytes.into()));
+            };
+            if let Err(err) = strip_concat_hash_bytes(cursor, hasher)? {
+                return Some(Err(err.into()));
+            }
+            match DecodedValueThunk::decode_with_metadata(cursor, *ty_id, metadata) {
+                Ok(decoded) => {
+                    match decoded.to_value() {
+                        Ok(value) => result.push(value.remove_context()),
+                        Err(err) => return Some(Err(err)),
+                    };
+                }
+                Err(err) => return Some(Err(err)),
+            }
+        }
+        Some(Ok(result))
+    }
+}
+
+/// Generates StorageKey implementations for tuples, e.g.
+/// ```rs,norun
+/// impl<A: EncodeAsType, B: EncodeAsType> StorageKey for (StorageKey<A>, StorageKey<B>) {
+///     fn keys_iter(&self) -> impl ExactSizeIterator<Item = &dyn EncodeAsType> {
+///         let arr = [&self.0 as &dyn EncodeAsType, &self.1 as &dyn EncodeAsType];
+///         arr.into_iter()
+///     }
+/// }
+/// ```
+macro_rules! impl_tuples {
+    ($($ty:ident $n:tt),+) => {{
+        impl<$($ty: EncodeAsType + ?Sized),+> StorageKey for ($( StaticStorageKey<$ty >),+) {
```

Comment:
Should this be more general and apply to any tuple whose members also impl `StorageKey`, rather than being restricted to tuples of `StaticStorageKey`?